libs/ui/src/themes/make_theme.ts

```diff
@@ -34,14 +64,123 @@ const colorThemes: Record<ColorMode, ColorTheme> = {
   legacy: COLOR_THEME_LEGACY,
 };
 
-// TODO: Actually implement these themes:
+const INCHES_PER_MM = 1 / 25.4;
+const PIXELS_PER_INCH = 72;
+
+function mmToPx(mm: number): number {
+  // TODO: This is just a stop-gap proxy - need to add device detection logic to
+  // determine what screen we're on (ELO/VSAP/laptop) and use pre-defined
+  // ratios, since we can measure physical display dimensions more accurately.
+  const { devicePixelRatio } = window;
+
+  return mm * INCHES_PER_MM * devicePixelRatio * PIXELS_PER_INCH;
+}
+
+function getFontSize(mode: SizeMode): number {
+  // Use the average midpoint value of the relevant VVSG size range.
+  const capitalLetterHeightMm =
+    (VVSG_CAPITAL_LETTER_HEIGHTS_MM[mode].min +
+      VVSG_CAPITAL_LETTER_HEIGHTS_MM[mode].max) /
+    2;
+
+  const fullFontHeightMm =
+    capitalLetterHeightMm * CAPITAL_HEIGHT_TO_FULL_FONT_HEIGHT_RATIO;
+
+  return mmToPx(fullFontHeightMm);
+}
+
 const sizeThemes: Record<SizeMode, SizeTheme> = {
-  s: { fontDefault: 10 },
-  m: { fontDefault: 14 },
-  l: { fontDefault: 18 },
-  xl: { fontDefault: 24 },
+  s: {
+    fontDefault: getFontSize('s'),
+    fontWeight: {
+      bold: 600,
+      light: 200,
+      regular: 300,
+    },
+    headingsRem: {
+      h1: 2.5,
+      h2: 1.75,
+      h3: 1.5,
+      h4: 1.25,
+      h5: 1.125,
+      h6: 1,
+    },
+    letterSpacingEm: 0.01,
```

Comment:
Why headings in `rem` and letter spacing in `em`?

Reply:
Making the headings use `rem` so that they have consistent sizing relative to the base font size, whereas using `em` for the letter spacing makes that scale proportionally to the size of the particular text/heading being rendered, rather than having one letter spacing value for all font sizes